src/global_scheduler/global_scheduler_algorithm.c

```diff
@@ -126,17 +127,18 @@ db_client_id get_photon_id(global_scheduler_state *state,
     return photon_id;
   }
 
-  /* Check to make sure this photon_db_client_id matches one of the schedulers*/
-  local_scheduler *lsptr = NULL;
-  for (lsptr = (local_scheduler *) utarray_front(state->local_schedulers);
-       lsptr != NULL; lsptr = (local_scheduler *) utarray_next(
-                          state->local_schedulers, lsptr)) {
-    if (memcmp(&lsptr->id, &photon_id, sizeof(photon_id)) == 0) {
+  /* Check to make sure this photon_db_client_id matches one of the
+   * schedulers. */
+  local_scheduler *local_scheduler_ptr;
+  for (int i = 0; i < utarray_len(state->local_schedulers); ++i) {
```

Comment:
actually this for loop change won't work. Subsequent if (!lsptr) check assumes that lsptr reaches NULL value if it finds no matching local schedulers after traversing the full array.